src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -487,7 +487,10 @@ public Guid Guid
         public virtual void Dispose()
         {
             foreach (var node in Nodes)
+            {
                 DisposeNode(node);
+                node.Dispose();
+            }
```

Comment:
Having a call `DisposeNode(node)` right before `node.Dispose()` surely is interesting, why isn't that second call inside `DisposeNode`?